code/game/jobs/role_authority.dm

```diff
@@ -259,8 +259,8 @@ I hope it's easier to tell what the heck this proc is even doing, unlike previou
 	var/list/roles_left = list()
 	var/assigned = 0
 	for(var/priority in HIGH_PRIORITY to LOW_PRIORITY)
-		// Assigning xenos first.
-		assigned += assign_initial_roles(priority, roles_for_mode & GLOB.ROLES_XENO, unassigned_players)
+		// Assigning xenos first, should not be counted
+		assign_initial_roles(priority, roles_for_mode & GLOB.ROLES_XENO, unassigned_players)
```

Comment:
What makes you think this? This contributes to the `GLOB.players_preassigned` when then is used to `set_spawn_positions` where the quantity of xenos is set to be 1:4. If say there were 100 players ready and it could assign all of them, `assign_initial_roles` should return the value 100. Now when you call `set_spawn_positions` for xeno jobs, that count gets multiplied against .25 meaning we ought to have 25 xenos. 25/100 is 1:4. If you instead don't count them, and lets say 25 would get assigned to xeno for `assign_initial_roles` then the count will be 75. 75 * .25 will be 18.75. So then `set_spawn_positions` will make 18/100 positions which is not 1:4.